llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -713,27 +713,43 @@ class Instruction : public sandboxir::User {
 #endif
 };
 
-class SelectInst : public Instruction {
-  /// Use Context::createSelectInst(). Don't call the
-  /// constructor directly.
-  SelectInst(llvm::SelectInst *CI, Context &Ctx)
-      : Instruction(ClassID::Select, Opcode::Select, CI, Ctx) {}
-  friend Context; // for SelectInst()
+/// Instructions that contain a single LLVM Instruction can inherit from this.
+class SingleLLVMInstructionImpl : public Instruction {
```

Comment:
It would be nice if we could go a little further and have `verify()` and `dump()` here as well to remove some more repetition.

The cast in `verify` should be doable using CRTP, but I don't have any good idea for the `"Expected SomeInst"` message that doesn't involve macros and operand stringification, which is a bit ugly.

Reply:
Good points, let me try to add both verify() and dump().

Reply:
`SingleLLVMInstructionImpl` is now a template that takes the llvm type as the argument, which is then used in verify(). I also removed the `dump()` functions. We now only have a non virtual `Value::dump()` that calls virtual `Value::dumpOS(raw_ostream &)` that gets overriden by the instructions. I had to rename `dump(raw_ostream &)` to `dumpOS(raw_ostream &)` because otherwise gdb would try to call the stream version of the dump and would not work with a simple `(gdb) call I->dump()`